lib/ts/translation/translationContext.tsx

```diff
@@ -29,27 +30,16 @@ export const TranslationContextProvider: React.FC<
     }>
 > = ({ children, defaultLanguage, userTranslationFunc, defaultStore, translationControlEventSource }) => {
     const [translationStore, setTranslationStore] = useState<TranslationStore>(defaultStore);
-    const cookieLang = getCurrentLanguageFromCookie();
-    const [currentLanguage, setCurrentLanguage] = useState(cookieLang === null ? defaultLanguage : cookieLang);
+    const [currentLanguage, setCurrentLanguage] = useState<string | undefined>(undefined);
 
-    const translateFunc = useCallback<TranslationFunc>(
-        (key: string) => {
-            if (userTranslationFunc !== undefined) {
-                return userTranslationFunc(key);
-            }
-            const res = translationStore[currentLanguage] && translationStore[currentLanguage][key];
-            const fallback = translationStore[defaultLanguage] && translationStore[defaultLanguage][key];
-
-            if (res === undefined) {
-                if (fallback !== undefined) {
-                    return fallback;
-                }
-                return key;
-            }
+    const loadLanguageFromCookies = useCallback(
+        async function () {
+            let cookieLang = await getCurrentLanguageFromCookie();
```

Comment:
I'd prefer if this were a single-line const assignment, but it's fine either way :)